File: 0061-rotate-list/0061-rotate-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(!head){
            return head;
        }
        int len = 1;
        ListNode* temp = head;
        while(temp->next)
        {
            temp = temp->next;
            len++;
        }
        k = len-(k%len);
        temp->next = head;
        while(k--)
        {
            temp = temp->next;
        }
        head = temp->next;
        temp->next = NULL;
        return head;
    }
};

File: 0061-rotate-list/0061-rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        lst_len = 1
        curr = head
        if not head:
            return head
        while curr.next:
            lst_len += 1
            curr = curr.next
        k = lst_len-(k%lst_len)
        curr.next = head
        while k > 0:
            curr = curr.next
            k -= 1
        head = curr.next
        curr.next = None
        return head





        